src/H5FDmpi.c

```diff
@@ -112,25 +146,43 @@ H5FD_mpi_get_size(const H5FD_t *file)
  * Programmer:	Quincey Koziol
  *              Friday, January 30, 2004
  *
+ * Changes:     Reworked function to use the ctl callback so we can get
+ *              rid of H5FD_class_mpi_t.  Since there are no real limits
+ *              on what the ctl callback can do, its file parameter can't
+ *              be constant.  Thus, I had to remove the const qualifier
+ *              on this functions file parameter as well.  Note also the
+ *              circumlocution required to use the ctl callbacks output
+ *              parameter to pass back the rank without introducing
+ *              compiler warnings.
+ *                                             JRM -- 8/13/21
+ *
  *-------------------------------------------------------------------------
  */
 MPI_Comm
-H5FD_mpi_get_comm(const H5FD_t *file)
+H5FD_mpi_get_comm(H5FD_t *file)
 {
-    const H5FD_class_mpi_t *cls;
-    MPI_Comm                ret_value;
+    const H5FD_class_t *cls;
+    uint64_t            flags    = H5FD_CTL__FAIL_IF_UNKNOWN_FLAG | H5FD_CTL__ROUTE_TO_TERMINAL_VFD_FLAG;
```

Comment:
I'm not a big fan of these two flags that need to be passed almost everywhere. Why not make it the default behavior? Also why the double __ after H5FD_CTL ?

Reply:
Actually, I have a use case I'll be working on shortly that doesn't use either of these flags.

As for the H5FD_CTL__, that is a convention I have been using for package specific flags since forever.  That said, I don't regard it as particularly important one way or the other.